Make optimizer docstrings consistent (#7891)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Documentation 📝</h3>

> * Docstrings for all optimizers were slightly refactored for consistency and legibility.

Reviewer: ```suggestion
* Slightly improved the docstrings of all optimizers for consistency and legibility.
```
